sdk/core/azure-core/test/ut/transport_policy_options.cpp

```diff
@@ -430,20 +430,41 @@ namespace Azure { namespace Core { namespace Test {
   TEST_F(TransportAdapterOptions, DisableCrlValidation)
   {
     Azure::Core::Url testUrl(AzureSdkHttpbinServer::Get());
-    //    Azure::Core::Url testUrl("https://www.microsoft.com/");
+
     // HTTP Connections.
+    auto failedCounter = 0;
+    auto const attempts = 3;
+    for (auto i = 0; i < attempts; i++)
     {
-      Azure::Core::Http::Policies::TransportOptions transportOptions;
+      GTEST_LOG_(INFO) << "DisableCrlValidation test iteration " << i << ".";
+      try
+      {
+        Azure::Core::Http::Policies::TransportOptions transportOptions;
 
-      // Note that the default is to *disable* CRL checks, because they are disabled
-      // by default. So we test *enabling* CRL validation checks.
-      transportOptions.EnableCertificateRevocationListCheck = true;
-      HttpPipeline pipeline(CreateHttpPipeline(transportOptions));
+        // Note that the default is to *disable* CRL checks, because they are disabled
+        // by default. So we test *enabling* CRL validation checks.
+        transportOptions.EnableCertificateRevocationListCheck = true;
+        HttpPipeline pipeline(CreateHttpPipeline(transportOptions));
 
-      auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, testUrl);
-      auto response = pipeline.Send(request, Azure::Core::Context::ApplicationContext);
-      EXPECT_EQ(response->GetStatusCode(), Azure::Core::Http::HttpStatusCode::Ok);
+        auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, testUrl);
+        auto response = pipeline.Send(request, Azure::Core::Context::ApplicationContext);
+        EXPECT_EQ(response->GetStatusCode(), Azure::Core::Http::HttpStatusCode::Ok);
+      }
+      catch (Azure::Core::Http::TransportException const& ex)
+      {
+        // CURL returns a connection error which triggers a transport exception.
+        GTEST_LOG_(INFO) << "DisableCrlValidation test iteration " << i
+                         << " failed with a TransportException.";
+        failedCounter++;
+        // We allow 1 intermittent failure, due to networking issues.
+        if (failedCounter > 1)
```

Comment:
You exit the loop on a single failure but line 467 checks for 3 failures.

That means that the assert at line 467 can never be hit.

Reply:
I exit the loop on the **second** failure. On a single success, failedCounter will be less than 3 attempts, and hence the assert will be hit and pass.

`1 > 1` is not true

Reply:
If this is a persistent failure, you will exit the loop with failedCounter set to 2 (because 2 > 1). At that point, `failedCounter` will be 2 and `attempts` will be 3.

2 < 3 is true.

Silly question: Why write your own implementation of a retry policy and not simply add a retry policy in the pipeline with a max retries set to 3.?

Reply:
At `failedCounter > 1`, we throw the exception that got us to the failure. So that assert isn't hit in the persistent failure case, and the test will exit on the throw.

> Silly question

I want to keep the test's intent, as is. As a *unit test*, focused on the transport policy.

Reply:
I don't see a PR blocking issue here.

If you have a specific suggestion to re-write it in a style you like, let me know :)

Reply:
I am not a fan of dead code in tests, or checks in tests that can never fail.

The newly added check at line 467 clearly falls into that category. If failedCounter is ever greater than 1, the test is going to throw an exception which should be caught by gtest and trigger a test failure.

There is no circumstance in which the check at line 467 will *ever* fail.

My specific suggestion for this test: Please remove this any other checks that have been added which can never fail. They provide no value to the test and potentially cause confusion in the future (when a future maintainer looks at the check and tries to figure out how what it is attempting to verify). 

And yes, I've *often* been in the shoes of the developer who has to reverse engineer a test to figure out what it's doing.

Reply:
> There is no circumstance in which the check at line 467 will *ever* fail.

Agreed, removed.